Added datetime util function

andysbolton opened the pull request:
Added a datetime parsing function that is hopefully broad enough to cover several different date strings. Currently returns a string formatted as yyyy-mm-dd. If it fails, it returns an empty string. Open to any suggestions on improving the robustness of the util. Here are some examples of tests (these all return 'OK'):

```
import unittest
from utils import parse_date_string

class DateTests(unittest.TestCase):
    def test_hyphen(self):
        self.assertEqual(parse_date_string("5-30-17"), "2017-05-30")

    def test_long(self):
        self.assertEqual(parse_date_string("05/23/2017 6:30 PM - 7:00 PM "), "2017-05-23")

    def test_brisbaneca(self):
        """test for sample from brisbaneca
        """
        self.assertEqual(parse_date_string("Thursday, December 28, 2017 - "), "2017-12-28")

    def test_danville(self):
        """test for sample from danville
        """
        self.assertEqual(parse_date_string("June 13, 2017 - 08:30 AM"), "2017-06-13")

    def test_losgatos(self):
        """test for sample from los gatos
        """
        self.assertEqual(parse_date_string("Jun  6, 2017"), "2017-06-06")

    def test_wacky(self):
        self.assertEqual(parse_date_string("this / one / wacky 17, ---/ test June 2033//"), "2033-06-17")

    # two dates in a string will always fail
    def test_fail(self):
        self.assertEqual(parse_date_string("June 13, 2017 and 5/17/98"), "")

if __name__ == '__main__':
    unittest.main()
```

andysbolton committed:
Added datetime util function

andysbolton committed:
Added datetime util function

bstarling commented:
Looks good & thanks for showing the tests, that is really helpful. We definitely need to starting adding tests to the project. (And would welcome addition of these to the PR) 

Just a couple of minor things:

- Since dateutil is an external package, can you add it to `requirements.txt` so it gets included in the project dependencies? 

- Sorry, I was not specific about this in the issue but I think we should return a `datetime.date` object instead of a string. The JSON exporter will just convert it to a string but other downstreams can directly handle python datetime objects. The `ca_dublin` spider does this in a (non generalizable) way.

Neither of this should stop the merge but I will leave the PR open a little longer incase you want to update it on this PR. Otherwise you can just make it in a future request.

Thanks again!

Dublin date functionality example:
```
In [8]: record_date = 'June 6, 2017'

In [9]: datetime.datetime.strptime(record_date, '%B %d, %Y').date()
Out[9]: datetime.date(2017, 6, 6)
```

andysbolton committed:
Add Fremont spider

* Added ca_fremont scrapy

* Moved base_url reference to top

* Added ca_fremont scrapy spider

* Changed Dublin class name to Fremont

* Accidentally removed backslash

Accidentally removed backslash from Belmont scraper, adding it back

andysbolton commented:
Sounds good! I'll make those changes soon.

andysbolton committed:
Modified parse_date_string() util

Changed parse_date_string() util function to return a date object instead of a formatted string. Also added dateutil dependency to the requirements.txt file.

andysbolton committed:
Modified parse_date_string() util

Changed parse_date_string() util function to return a date object instead of a formatted string. Also added dateutil dependency to the requirements.txt file.

andysbolton committed:
Merge branch 'datetime' of https://github.com/andysbolton/town-council into datetime

andysbolton committed:
Fixed package name

Fixed package name from 'dateutil' to 'python-dateutil'
